jep/README.md

```diff
@@ -0,0 +1,202 @@
+```
+
+A base widget toolkit is in-scope of this JEP. Reactivity and collaborative are out-of-scope of this JEP.
+
+It should be possible for a React.js widget to share a global state with a widget developed in another technology (Vue.js, Svelte...). This can be achieved by a adequate global state/store system that is not restricted a single technology (so should be `polyglot`). This can be defined in subsequent JEP.
+
+Reactivity to support features like [Jupyter Widgets](https://github.com/jupyter-widgets) can be defined in subsequent JEP. The current implementation of Jupyter Widgets relies on [Backbone.js](https://backbonejs.org) for which a successor would be welcome. We could imagine relying on the collaborative feature based on CRDT (Conflict-free replicated data types).
+
+Realtime collaboration and local-first aplication are not in-scope for now but it is logical to mention that and depend on a CRDT solution (Y.js in use for now).
+
+Integration with the current Lumino extension system could be considered in the future. As this is a bit more complex to design, especially to support multiple technologies (not only React.js), we think a separated discussion will be better for that.
+
+There are also ongoing discussion to add more React.js and Web Component features in the Lumino toolkit, these topics should be discussed in their respective issues:
+
+- Lumino v2 Plan https://github.com/jupyterlab/lumino/issues/233
+- Switch from widget to standard web components https://github.com/jupyterlab/lumino/issues/290
+
+# Future possibilities
+
+```
+Think about what the natural extension and evolution of your proposal would
+be and how it would affect the Jupyter community at-large. Try to use this section as a tool to more fully consider all possible
+interactions with the project and language in your proposal.
+Also consider how the this all fits into the roadmap for the project
+and of the relevant sub-team.
+
+This is also a good place to "dump ideas", if they are out of scope for the
+JEP you are writing but otherwise related.
+
+If you have tried and cannot think of any future possibilities,
+you may simply state that you cannot think of anything.
+
+Note that having something written down in the future-possibilities section
+is not a reason to accept the current or a future JEP; such notes should be
+in the section on motivation or rationale in this or subsequent JEPs.
+The section merely provides additional information.
+```
+
+As mentioned earlier, subsequent JEP are expected to answer the open questions.
+
+The toolkit will target basic visual components and will not deliver higher level components like a usable `Cell`, or a collection of cells showing a basic `Notebook` connected to a running Kernel. Such components are for example available in the [Jupyter React](https://github.com/datalayer/jupyter-react) and would demeand further discussion around global state, connection to the services hosted on a server...
```

Comment:
The interacdtion with Lumino and how it has to be taken into account is indeed to be clarified. In terms of widget, this is probably not linked. In terms of signals, I can see a global state binding to the lumino signals, but this would be for subsequent JEP.

Reply:
To precise my thought about widgets, I would like to mention in the potential follow up that 

> the technology put in place for this JEP once matured could be used to migrate the current Lumino widgets to ease their integration into modern frameworks

Reply:
Committed in https://github.com/datalayer/jupyter-react/commit/02538c97e07135b244bfda02d0567b18e97a8d14. Resolvoing.